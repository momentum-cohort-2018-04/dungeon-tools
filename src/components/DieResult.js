import React from 'react'

import d4 from '../img/d4.png'
import d6 from '../img/d6.png'
import d8 from '../img/d8.png'
import d10 from '../img/d10.png'
import d12 from '../img/d12.png'
import d20 from '../img/d20.png'
import dice from '../img/dice.png'

function dieImg (size) {
  if (size === 4) {
    return d4
  } else if (size === 6) {
    return d6
  } else if (size === 8) {
    return d8
  } else if (size === 10) {
    return d10
  } else if (size === 12) {
    return d12
  } else if (size === 20) {
    return d20
  } else {
    return dice
  }
}

function DieResult (props) {
  const {sides, roll} = props
  return (
    <div className='DieResult'>
      <img src={dieImg(sides)} style={{width: '100%'}} />
      <div>{roll}</div>
    </div>
  )
}

export default DieResult
