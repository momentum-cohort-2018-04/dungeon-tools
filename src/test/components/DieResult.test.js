/* globals it, test, expect */

import React from 'react'
import ReactDOM from 'react-dom'
import DieResult from '../../components/DieResult'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'

// Smoke test

test('renders without crashing', () => {
  const div = document.createElement('div')
  ReactDOM.render(<DieResult sides={6} roll={4} />, div)
  ReactDOM.unmountComponentAtNode(div)
})

// Snapshot test

test('renders to a snapshot', () => {
  const component = renderer.create(
    <DieResult sides={6} roll={4} />
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

// Enzyme test
test('renders an image and shows the result', () => {
  const wrapper = shallow(<DieResult sides={6} roll={4} />)
  expect(wrapper.find('img')).toHaveLength(1)
  expect(wrapper.text()).toContain('4')
})
