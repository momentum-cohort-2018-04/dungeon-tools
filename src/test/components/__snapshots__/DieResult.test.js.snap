// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders to a snapshot 1`] = `
<div
  className="DieResult"
>
  <img
    src="d6.png"
    style={
      Object {
        "width": "100%",
      }
    }
  />
  <div>
    4
  </div>
</div>
`;
